cortex/tsdb.libsonnet

```
@@ -222,7 +222,17 @@
       'blocks-storage.bucket-store.index-header-lazy-loading-enabled': 'true',
       'blocks-storage.bucket-store.index-header-lazy-loading-idle-timeout': '60m',
 
+      local indexMaxConcurrency = 100,
+      local chunksMaxConcurrency = 100,
+      local metaMaxConcurrency = 100,
       'blocks-storage.bucket-store.max-chunk-pool-bytes': 12 * 1024 * 1024 * 1024,
+      'blocks-storage.bucket-store.index-cache.memcached.max-get-multi-concurrency': indexMaxConcurrency,
+      'blocks-storage.bucket-store.chunks-cache.memcached.max-get-multi-concurrency': chunksMaxConcurrency,
+      'blocks-storage.bucket-store.metadata-cache.memcached.max-get-multi-concurrency': metaMaxConcurrency,
+      'blocks-storage.bucket-store.index-cache.memcached.max-idle-connections': indexMaxConcurrency,
+      'blocks-storage.bucket-store.chunks-cache.memcached.max-idle-connections': chunksMaxConcurrency,
+      'blocks-storage.bucket-store.metadata-cache.memcached.max-idle-connections': metaMaxConcurrency,
```

We want to make it clear one depends on the other one. Also if someone overrides `max-get-multi-concurrency` then `max-idle-connections` should be updated automatically.

```suggestion
    'blocks-storage.bucket-store.index-cache.memcached.max-idle-connections': $.store_gateway_args['blocks-storage.bucket-store.index-cache.memcached.max-get-multi-concurrency'],
    'blocks-storage.bucket-store.chunks-cache.memcached.max-idle-connections': $.store_gateway_args['blocks-storage.bucket-store.chunks-cache.memcached.max-get-multi-concurrency'],
    'blocks-storage.bucket-store.metadata-cache.memcached.max-idle-connections': $.store_gateway_args['blocks-storage.bucket-store.metadata-cache.memcached.max-get-multi-concurrency'],
```

---

@pracucci aha, now I understand the reason for the previous indirection. Thanks, will restore it.